project/UI/Jekyll/Jekyll.swift

```diff
@@ -2,51 +2,50 @@
-            let mainPage = mainPagePath == "README.md" ? output.appendingPathComponent("README.md") : URL(fileURLWithPath: mainPagePath, isDirectory: false)
-            
-            return IO.pure((input: input, output: output, mainPage: mainPage))
-        }^
-    }
+    @ArgumentParser.Option(help: "Path to the Xcode Playground to render")
+    var project: String
+    
+    @ArgumentParser.Option(help: "Path where the resulting jekyll files will be generated")
+    var output: String
+    
+    @ArgumentParser.Option(name: .customLong("main-page"), default: "README.md", help: "Path to 'README.md' file to be used as the index page")
+    var mainPage: String
     
-    let console = Console(script: script,
-                          description: "Render markdown files that can be consumed from Jekyll to generate a microsite",
-                          arguments: .init(name: JekyllCommand.project.rawValue, placeholder: "path-to-input", description: "path to the nef Playground to render"),
-                                     .init(name: JekyllCommand.output.rawValue, placeholder: "path-to-output", description: "path where the resulting Markdown files will be generated"),
-                                     .init(name: JekyllCommand.main.rawValue, placeholder: "path-to-index", description: "path to 'README.md' file to be used as the index page", default: "README.md"))
+    var projectURL: URL { URL(fileURLWithPath: project.trimmingEmptyCharacters.expandingTildeInPath) }
+    var outputURL: URL  { URL(fileURLWithPath: output.trimmingEmptyCharacters.expandingTildeInPath) }
+    var mainPath: String { mainPage.trimmingEmptyCharacters.expandingTildeInPath }
     
-    return arguments(console: console)
-        .flatMap { (input, output, mainPage) in
-            nef.Jekyll.render(playgroundsAt: input, mainPage: mainPage, into: output)
-                      .provide(console)^
-                      .mapError { _ in .render() }
-                      .foldM({ _ in console.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
-                             { _ in console.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   }) }^
-        .reportStatus(in: console)
-        .foldM({ e in console.exit(failure: "\(e)")        },
-               { success in console.exit(success: success) })
-        .unsafeRunSyncEither()
+    
+    public func main() -> IO<CLIKit.Console.Error, Void> {
+        arguments(parsableCommand: self)
+            .flatMap { (input, output, mainPage) in
+                nef.Jekyll.render(playgroundsAt: input, mainPage: mainPage, into: output)
+                    .provide(Console.default)^
+                    .mapError { _ in .render() }
+                    .foldM({ _ in Console.default.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
+                           { _ in Console.default.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   })
```

Comment:
If you don't care about the error in the output, why mapping it? Review all scripts, because you either have it fixed to a concrete value (therefore no need to map it) or you don't even read it.

Reply:
In the `UI` (CLI) - the output is not important because during the execution of the API we are showing information about the process, but we need to know if it goes from error/valid path in order to `exit` success or failure - `.exit(failure:)` and `.exit(success:)` right?

Reply:
What I mean is that you are using `mapError` above but then ignoring its value in `foldM`. Are you using this mapped error later at any point?

Reply:
Nop, it is tricky to make happy the compiler `Console` works with `Console.Error` but it receives another kind of Error.